backend/compact-connect/tests/app/test_pipeline.py

```diff
@@ -58,29 +58,92 @@ def test_synth_pipeline(self):
                 persistent_stack_template.find_resources(CfnUserPoolResourceServer.CFN_RESOURCE_TYPE_NAME),
             )
-            # Ensure the resource servers are created with the expected scopes
+            # Ensure the compact resource servers are created with the expected scopes
             self.assertEqual(
                 ['admin', 'write', 'readGeneral', 'readSSN'],
                 [scope['ScopeName'] for scope in resource_server_properties['Scopes']],
                 msg=f'Expected scopes for compact {compact} not found',
             )
 
+    def test_synth_generates_compact_resource_servers_with_expected_scopes_for_staff_users_test_stage(self):
+        persistent_stack = self.app.pipeline_stack.test_stage.persistent_stack
+        self._when_testing_compact_resource_servers(persistent_stack, environment_name='test')
+
+    def test_synth_generates_compact_resource_servers_with_expected_scopes_for_staff_users_prod_stage(self):
+        persistent_stack = self.app.pipeline_stack.prod_stage.persistent_stack
+        self._when_testing_compact_resource_servers(persistent_stack, environment_name='prod')
+
+    def _when_testing_jurisdiction_resource_servers(self, persistent_stack, snapshot_name, overwrite_snapshot):
+        persistent_stack_template = Template.from_stack(persistent_stack)
+
+        # Get the jurisdiction resource servers created in the persistent stack
+        resource_servers = persistent_stack.staff_users.jurisdiction_resource_servers
+        # We must confirm that these scopes are being explicitly created for each active jurisdiction
+        # which are absolutely critical for the system to function as expected.
+        # If a new jurisdiction is made active within the system, this test will need to be updated
+        jurisdiction_resource_server_config = []
+        for _jurisdiction, resource_server in resource_servers.items():
+            resource_server_properties = self.get_resource_properties_by_logical_id(
+                persistent_stack.get_logical_id(resource_server.node.default_child),
+                persistent_stack_template.find_resources(CfnUserPoolResourceServer.CFN_RESOURCE_TYPE_NAME),
+            )
+            # remove dynamic user pool id
+            del resource_server_properties['UserPoolId']
+            jurisdiction_resource_server_config.append(resource_server_properties)
+
+        # sort the resource server list by jurisdiction for consistency
+        jurisdiction_resource_server_config.sort(key=lambda jurisdiction: jurisdiction['Identifier'])
+        # sort the scopes within the resource server by name for consistency
+        for resource_server in jurisdiction_resource_server_config:
+            resource_server['Scopes'].sort(key=lambda scope: scope['ScopeName'])
+        # this will only include resource server scopes for compacts/jurisdictions that are marked as active
+        # for the test environment
```

Comment:
can't this be the prod environment, too? Should we change to–

```suggestion
        # for the environment
```